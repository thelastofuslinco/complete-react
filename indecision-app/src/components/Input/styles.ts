import styled from 'styled-components'

export const InputContainer = styled.input`
  padding: 1rem;
  border: none;
  border-bottom: 1px solid black;
`
