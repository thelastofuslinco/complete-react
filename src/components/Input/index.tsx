import { Component, InputHTMLAttributes } from 'react'

interface State {
  value: string | number | readonly string[]
}

interface Props extends InputHTMLAttributes<HTMLInputElement> {}

class Input extends Component<Props, State> {
  state = { value: this.props.value || '' }

  render() {
    return (
      <input
        {...this.props}
        className={`input ${this.props.className}`}
        value={this.state.value}
        onChange={(event) => this.setState({ value: event.target.value })}
      />
    )
  }
}

export default Input
