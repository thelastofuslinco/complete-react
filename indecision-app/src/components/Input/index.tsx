import { InputContainer } from './styles'

const Input = (props: React.InputHTMLAttributes<HTMLInputElement>) => {
  return <InputContainer {...props} />
}

export default Input
