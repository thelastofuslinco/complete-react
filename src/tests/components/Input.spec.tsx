import { screen, render, fireEvent } from '@testing-library/react'
import Input from '../../components/Input'

describe('Input Component', () => {
  test('should render and change the Input component', async () => {
    render(<Input type="text" />)
    const input: HTMLInputElement = screen.getByRole('textbox')
    fireEvent.change(input, { target: { value: 'any_value' } })
    expect(input.value).toBe('any_value')
    expect(input).toMatchSnapshot()
  })
})
