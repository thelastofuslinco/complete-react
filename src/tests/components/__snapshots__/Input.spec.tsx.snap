// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component should render and change the Input component 1`] = `
<input
  class="input undefined"
  type="text"
  value="any_value"
/>
`;
